library/ssl_cli.c

```diff
@@ -3006,8 +3006,8 @@ static int ssl_parse_server_key_exchange( mbedtls_ssl_context *ssl )
             mbedtls_ssl_send_alert_message(
                 ssl,
                 MBEDTLS_SSL_ALERT_LEVEL_FATAL,
-                MBEDTLS_SSL_ALERT_MSG_ILLEGAL_PARAMETER );
-            return( MBEDTLS_ERR_SSL_BAD_HS_SERVER_KEY_EXCHANGE );
+                MBEDTLS_SSL_ALERT_MSG_DECODE_ERROR );
```

Comment:
Looking at the possible cause of this alert, it can happen when ssl_parse_server_psk_hint tries to decode some data that is too short, i.e. it's more accurate to use DECODE_ERROR ("message could not be parsed due to a syntactic error") rather than ILLEGAL_PARAMETER ("A field in a message was incorrect or inconsistent with other fields"). So I think it's the right choice. I will separate this out.

Reply:
Now separated, new commit de128d502f is an exact match against the reviewed version 3a1f2c350d6.